src/main/java/com/google/dicomwebfuse/dao/FuseDaoImpl.java

```diff
@@ -382,9 +386,11 @@ private void createRequestForUploadInstance(URIBuilder uriBuilder, Path instance
       try (CloseableHttpResponse response = httpclient.execute(request)) {
         int statusCode = response.getStatusLine().getStatusCode();
         if (statusCode != HttpStatusCodes.STATUS_CODE_OK) {
+          Document parsedXmlResponse = Jsoup
+              .parse(EntityUtils.toString(response.getEntity()), "", Parser.xmlParser());
```

Comment:
So unfortunately not all responses will be in XML format... Anything that gets to the DICOMweb portion will be XML, but anything at the Healthcare API portion (e.g. permission denied, missing dicom store) will be JSON. Can we handle that here and add a unit tests for both of these cases?

Reply:
should json be formatted as well?

Reply:
I believe the JSON is already nicely formatted.

Reply:
done